datadog-setup.php

```diff
@@ -400,6 +701,118 @@ function ($el) {
+ * files are split by SAPI, so we try and add the Apache SAPI INI files as
+ * well. In case it exists, we also add the default `php.ini` file to the list.
+ *
+ * The returned array is somewhat sorted to have the "default" INI file for
+ * Datadog (`98-ddtrace.ini`) in the beginning of the array.
+ *
+ * @see ini_values
+ * @return string[]
+ */
+function find_all_ini_files(array $phpProperties): array
+{
+    $iniFilePaths = [];
+
+    $addIniFiles = function (string $path) use (&$iniFilePaths) {
+        if (!is_dir($path)) {
+            return;
+        }
+        foreach (scandir($path) as $ini) {
+            if (!is_file($path . '/' . $ini) || substr($ini, -4) !== '.ini') {
+                continue;
+            }
+            $iniFile = $path . '/' . $ini;
+            if (strpos($ini, '98-ddtrace.ini') !== false) {
+                array_unshift($iniFilePaths, $iniFile);
+            } else {
+                $iniFilePaths[] = $iniFile;
+            }
+        }
+    };
+
+    if ($phpProperties[INI_SCANDIR]) {
+        $addIniFiles($phpProperties[INI_SCANDIR]);
+
+        if (strpos($phpProperties[INI_SCANDIR], '/cli/conf.d') !== false) {
+            /* debian based distros have INI folders split by SAPI, in a predefined way:
+             *   - <...>/cli/conf.d       <-- we know this from php -i
+             *   - <...>/apache2/conf.d   <-- we derive this from relative path
+             *   - <...>/fpm/conf.d       <-- we derive this from relative path
+             */
+            $apacheConfd = str_replace('/cli/conf.d', '/apache2/conf.d', $phpProperties[INI_SCANDIR]);
+            $addIniFiles($apacheConfd);
+        }
+    }
+
+    if ($phpProperties[INI_MAIN] && is_file($phpProperties[INI_MAIN])) {
+        $iniFilePaths = [$phpProperties[INI_MAIN]];
+    }
+
+    return $iniFilePaths;
+}
+
+/**
+ * This function will find the main INI file(s) for the given `$phpProperties`.
+ *
+ * The "main" or "default" INI file is either a `98-ddtrace.ini` file or
+ * another INI file that loads the extension (or rephrasing this as: the file
+ * the has the `extension = ddtrace` line in it).
+ *
+ * In most cases this function will return an array with exactly one element,
+ * only in case we detect a Debian based distribution, it might contain two
+ * elements, as debian based distributions split the INI folders based on SAPI
+ * and we include those as well.
+ *
+ * The `$phpProperties` can be retrieved by calling `ini_values($pathToPHPBinary)`.
+ *
+ * @see ini_values
+ * @return string[]
+ */
+function find_main_ini_files(array $phpProperties): array
+{
+    $iniFilePaths = [];
+    if ($phpProperties[INI_SCANDIR]) {
+        $iniFileName = '98-ddtrace.ini';
+        // Search for pre-existing files with extension = ddtrace.so to avoid conflicts
+        // See issue https://github.com/DataDog/dd-trace-php/issues/1833
+        foreach (scandir($phpProperties[INI_SCANDIR]) as $ini) {
```

Comment:
If `scandir` fails, such as:

```
% php --ini
Configuration File (php.ini) Path: /opt/php/8.2/etc/php.ini
Loaded Configuration File:         (none)
Scan for additional .ini files in: /opt/php/8.2/etc/conf.d
Additional .ini files parsed:      (none)
```

And that directory `/opt/php/8.2/etc/conf.d` doesn't exist`, then we fail.

Reply:
I added a few things to handle this better. If the directory doesn't exist, it will tell you to make it and try again.